### ISSUE TITLE
Franz prevents display sleep on OS X

### ISSUE THREAD

**Issue by Minkey27** [meetfranz/franz#554]
Some situations arise where Franz.app prevents your display from going to sleep. 
Running 'pmset -g assertions' in the terminal gives something like this:
pid 97182(Franz): [0x00006d300005956b] 00:03:23 NoDisplaySleepAssertion named: "Playing video"

Further monkey testing made me realise that telegram has gifs.The gifs in telegram play endlessly and gifs trigger the assertion mentioned above.

Even when you switch services, the gifs keep playing. So this causes your OS X device never to sleep. This however does not persist on quiting and restarting the franz.app. 

### Expected Behavior

So a solution for this: Stop all media when you switch conversations/services.

### Steps to Reproduce (for bugs)

1. Make a telegram service
2. Send a gif (https://i.imgur.com/P7RiAtO.mp4 <- for convenience)
3. Play it
4. Switch to another service and wait (maybe check pmset -g assertions)

### Your Environment

* Franz Version used: 5.0.0-beta.14
* Operating System and version: OS X 10.13.2

**Comment by svenjacobs** [comment]
I noticed the same issue! While a Telegram video (endless GIF) is played, even in the background, Franz prevents the computer from sleeping.

**Comment by dpinol** [comment]
I love Franz, but this is a deal-breaker. It drains my battery

**Comment by xbiggyl** [comment]
My Macbook pro 2016 on Mojave, never goes to sleep if Franz is running.
Battery drain is a deal breaker for me as well.
Sometimes if I forget to close franz before putting my mac to sleep overnight, the battery drains completely.

**Comment by AnthoPakPak** [comment]
I was also facing this issue. I've manage to avoid this behavior by performing "Reload Franz" shortcut (MAJ + CMD + R) every time before going to sleep. This way, it will "kill" every video that was launched through one the services. 

Of course, I'm not doing it manually each time. As I've already have BetterTouchTool app, I've set up following action (in Other tab) : 

"Before Mac Goes To Sleep" > Send Shortcut to Specific App (Franz) : `MAJ + CMD + R`

I've not seen any battery drain issue since ! I hope it can help some of you ;)

**Comment by ctheune** [comment]
Actually this is a security problem for me. We use "sleep after X minutes" and "lock password immediately after sleep" so that if I forget to lock my machine it will at least lock after a short period. There is no way to notice that Franz is doing this and actually is a complete showstopper as it breaks a very core assumption about how I can securely interact with a MacOS machine.

**Comment by briankendall** [comment]
I just experienced the same problem. It looks like in my case the culprit was one of those "new features" pop-ups with animated graphics that occasionally appears when Discord first loads. It seems their animated graphic counted as a video that was playing back.

**Comment by crjherbert** [comment]
This happens with Windows 10 also.

**Comment by Rvervuurt** [comment]
This is still a problem. I changed to Ferdi (a fork of Franz?) in hopes that it'd be fixed, but my computer has been running (screen on) all weekend...

**Comment by Julian88Tex** [comment]
May be related: https://github.com/getferdi/ferdi/issues/1055 & https://github.com/getferdi/ferdi/issues/776

**Comment by mastercko** [comment]
This is still happening and I believe it's related to auto-playing videos in various chat services (thanks to @briankendall 's comment I was able to eventually figure out what was happening in my case). My story below (x-posted from  getferdi/ferdi#1055):

---

I think I just figured out what may be going on. If there is a video playing in any of the chat services, it will hold a Wake Lock. I disabled all my services and found that the PC went to sleep as expected even with Ferdi running. Going through and reenabling each service one-by-one, there was a Facebook Messenger chat that I had been looking at with a video on that was shared by someone in the chat. Facebook Messenger, in its infinite wisdom, auto-plays any video as soon as it is in view (which happened on load for this particular chat in my case -- the video is before the fold in the first chat in my list). Upon pausing the video, the PC sleeps fine. So it seems like that's what's going on.

**Short version: Check through all your enabled services for running videos!**

For now, I just archived the chats in question, since I don't really need to see them or the videos in them anymore at the moment.

To the Ferdi/Franz team, something should be done to ensure background video playing like this (especially on Facebook Messenger) doesn't hold a wake lock. Feels like it should be more along the lines of isVideoPlaying && isInFocus is needed to hold a wake lock on (if only on the Facebook service, since it is so aggro with playing videos even if you don't actually want to see them).
